File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/EncryptedCredentialsFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.NotNullFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface EncryptedCredentialsFilter {

    AnnotationState value() default AnnotationState.ON;

    NotNullFilter data() default @NotNullFilter(AnnotationState.OFF);

    NotNullFilter hash() default @NotNullFilter(AnnotationState.OFF);

    NotNullFilter secret() default @NotNullFilter(AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/EncryptedCredentialsFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.NotNullFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class EncryptedCredentialsFilter(
    val status: AnnotationState = AnnotationState.ON,
    val validator: Array<String> = [],
    val data: NotNullFilter = NotNullFilter(
        status = AnnotationState.OFF
    ),
    val hash: NotNullFilter = NotNullFilter(
        status = AnnotationState.OFF
    ),
    val secret: NotNullFilter = NotNullFilter(
        status = AnnotationState.OFF
    )
)
